react-select/react-select.d.ts

```diff
@@ -1,80 +1,226 @@
+         * @default "No results found"
+         */
+        noResultsText?: string;
+        onBlur?: __React.FocusEventHandler;
+        /**
+         * whether to clear input on blur or not
+         * @default true
+         */
+        onBlurResetsInput?: boolean;
+        onChange?: (newValue: Option | Option[]) => void;
+        onFocus?: __React.FocusEventHandler;
+        onInputChange?: (inputValue: string) => void;
+        onOptionLabelClick?: (value: string, event: Event) => void;
+        /**
+         * function which returns a custom way to render the options in the menu
+         */
+        optionRenderer?: () => void;
+        /**
+         * array of Select options
+         * @default false
+         */
+        options?: Array<Option>;
+        /**
+         * field placeholder, displayed when there's no value
+         * @default "Select..."
+         */
+        placeholder?: string;
+        /**
+         * whether to enable searching feature or not
+         * @default true;
+         */
+        searchable?: boolean;
+        /**
+         * message to display whilst options are loading via asyncOptions, or when `isLoading` is true
+         * @default "Searching..."
+         */
+        searchingText?: string;
+        /**
+         * label to prompt for search input
+         * @default "Type to search"
+         */
+        searchPromptText?: string;
+        /**
+         * initial field value
+         */
+        value?: Option | Option[];
+        /**
+         * the option property to use for the value
+         * @default "value"
+         */
+        valueKey?: string;
+        /**
+         * function which returns a custom way to render the value selected
+         * @default false
+         */
+        valueRenderer?: () => void;
     }
 
-    interface ReactAsyncSelectProps extends React.Props<ReactSelectClass>{
-        cache? : any;
-        loadOptions? : ()=>void;
-        ignoreAccents? : boolean;
-        isLoading? : boolean;
-        loadingPlaceholder? : string;
+    interface ReactAsyncSelectProps extends __React.Props<ReactSelect> {
+        cache?: any;
+        loadOptions?: () => void;
+        ignoreAccents?: boolean;
+        isLoading?: boolean;
+        loadingPlaceholder?: string;
     }
 
-    interface ReactSelect extends  React.ReactElement<ReactSelectProps> { }
-    interface ReactSelectClass extends React.ComponentClass<ReactSelectProps>{}
-
-    interface SelectWithAsync extends ReactSelectClass {
-        Async? : ReactAsyncSelectProps;
+    interface ReactSelect extends  __React.ReactElement<ReactSelectProps> { }
+    interface ReactSelectClass extends __React.ComponentClass<ReactSelectProps> {
+        Async: __React.ComponentClass<ReactAsyncSelectProps>;
     }
+}
 
-    var ReactSelect: SelectWithAsync;
-
-    export default ReactSelect;
+declare module "react-select" {
+    var select: ReactSelect.ReactSelectClass;
+    export = select;
```

Comment:
the packaged `react-select.js` file that i installed from bower exposes a single global `Select` which is the JSX constructor itself. There's no `default` export available, so I had to refactor typings to use the namespace syntax for them to compile. Perhaps if we were installing from npm and bundling react-select in webpack then this would be different.

Which is a pretty bad situation for typings, if the usage is wildly different based on your build system...